schema/global/rooms.ts

```diff
@@ -172,3 +189,108 @@ export const RoomKindQuery = extendType({
 		t.crud.roomKinds({ filtering: true });
 	},
 });
+
+const roomFieldsType = {
+	name: stringArg(),
+	kind: stringArg(),
+	vol: intArg(),
+	vent: stringArg(),
+	units: list(UnitMutationType)
+};
+
+export const RoomStatus = mutationStatusType({
+	name: "RoomStatus",
+	definition(t) {
+		t.string('name', { description: `A string representation of the new Room's object identity; may be thereafter passed to e.g. \`updateRoom\``});
+	}
+});
+
+export const RoomMutations = extendType({
+	type: 'Mutation',
+	definition(t) {
+		t.nonNull.field('updateRoom', {
+			description: `Update room details.`,
+			args: {
+				...roomFieldsType
+			},
+			type: "RoomStatus",
+			async resolve(root, args, context) {
+				const prisma = context.prisma;
+				return await prisma.$transaction(async (tx) => {
+					const room = await tx.Room.findFirst({ where: { name: args.name }});
+
+					if (room) {
+						const kindDesignation = await tx.RoomKind.findFirst({where: {name: args.kind}})
+						try {
+							const updatedRoom = await tx.Room.update(
+								{ where: { id: room.id },
+									data: {
+										vol: args.vol,
+										vent: args.vent,
+										kind: { connect: { id_labType: kindDesignation.id_labType}},
+									}
+								});
+
+							if (updatedRoom) {
+								if (args.units.length>0) {
+									const errors: string[] = [];
+									for (const addUnit of args.units) {
+
+										const unit = await tx.Unit.findFirst({ where: { name: addUnit.name }});
+
+										if (unit) {
+											if (addUnit.status == 'New') {
+												try {
+													const u = await tx.unit_has_room.create({
+														data: {
+															id_lab: room.id,
+															id_unit: unit.id
+														}
+													})
+													if ( !u ) {
+														errors.push(`Error creating unit ${unit.name}.`);
+													}
+												} catch ( e ) {
+													errors.push(`Error creating unit ${unit.name}.`);
+												}
+											}
+											else if (addUnit.status == 'Deleted') {
+												try {
+													const u = await tx.unit_has_room.deleteMany({
+														where: {
+															id_lab: room.id,
+															id_unit: unit.id
+														},
+													});
+													if (!u) {
+														errors.push(`Error deleting ${unit.name}.`);
+													}
+												} catch ( e ) {
+													errors.push(`Error creating unit ${unit.name}.`);
+												}
+											}
+										} else {
+											errors.push(`Unit ${addUnit.name} not found.`);
+										}
+									}
+
+									if (errors.length > 0) {
+										return mutationStatusType.error(`${errors.join('\n')}`);
+									} else {
+										return mutationStatusType.success();
```

Comment:
Same: only this code path should commit the transaction.

Reply:
Same as above